src/Plugin/Field/FieldFormatter/ErrandServicesFormatter.php

```diff
@@ -0,0 +1,131 @@
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * Gets the channel types.
+   *
+   * @return \Drupal\helfi_tpr\Entity\ChannelTypeCollection
+   *   The channel type collection.
+   */
+  private function getChannelTypes() : ChannelTypeCollection {
+    return ChannelTypeCollection::createFromArray($this->getSetting('sort_order'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $form = parent::settingsForm($form, $form_state);
+
+    $form['sort_order'] = [
+      '#type' => 'table',
+      '#caption' => $this->t('Order'),
+      '#header' => [
+        $this->t('ID'),
+        $this->t('Label'),
+        $this->t('Weight'),
+      ],
+      '#tableselect' => FALSE,
+      '#tabledrag' => [
+        [
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => 'group-weight',
+        ],
+      ],
+    ];
+
+    foreach ($this->getChannelTypes() as $item) {
+      $form['sort_order'][$item->id] = [
+        '#attributes' => ['class' => ['draggable']],
+        '#weight' => $item->weight,
+        'id' => ['#plain_text' => $item->id],
+        'label' => ['#plain_text' => $item->label()],
+        'weight' => [
+          '#type' => 'weight',
+          '#title' => $this->t('Weight for @title', ['@title' => $item->label()]),
+          '#title_display' => 'invisible',
+          '#default_value' => $item->weight,
+          '#attributes' => ['class' => ['group-weight']],
+        ],
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) : array {
+
+    if (!$items->getEntity() instanceof Service) {
+      throw new \InvalidArgumentException('The field can only be used with tpr_errand_service entities.');
+    }
+    $elements = parent::viewElements($items, $langcode);
+    $channelTypes = $this->getChannelTypes();
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+    foreach ($elements as $delta => $element) {
+      $elements[$delta]['unique_channels'] = [];
+      /** @var \Drupal\helfi_tpr\Entity\ErrandService $entity */
+      $entity = $element['#tpr_errand_service'];
+      foreach ($entity->getChannels() as $channel) {
+        if (in_array($channel->getType(), $elements[$delta]['unique_channels'])) {
+          continue;
+        }
+        $translatedChannel = $channel->getTranslation($language);
+
+        $elements[$delta]['unique_channels'][] = [
+          '#name' => $translatedChannel->type_string->value,
+          '#weight' => $channelTypes[$channel->getType()]->weight,
+        ];
+
+      }
+      if ($items->getEntity()->has_unit->value) {
+        $elements[$delta]['unique_channels'][] = [
+          '#name' => $this->t('Office'),
+          '#weight' => 999,
+        ];
+      }
+      uasort($elements[$delta]['unique_channels'], [SortArray::class, 'sortByWeightProperty']);
```

Comment:
see above comment, don't use the parent::viewElements($items, $langcode); the return item should be elements[0] with and item_list theme.. j